### PR TITLE
#1349, refactoring and enhancement of the API log configuration system

### LINKED ISSUE #1349: What about add create/update/delete actions for System::ApiLogConfig index page
## Description:

To have the ability to enable debug mode for the specific controllers and view more data in `Log::ApiLog` I suggest to add `create/update/delete` actions for `Api Log Configs` index page.

After these improvements, we have the ability to filter API Log by adding a controller to 'sys.api_log_config' table.

## Additional info:

When debug mode is enabled for specific controller ensure that additional info is stored to API log without any internal server errors.

## Related files:
- app/admin/system/api_log_configs.rb
- app/models/system/api_log_config.rb
- config/initializers/instrumentation_notification.rb

### PR DESCRIPTION
**Pull request by Ivanov-Anton**
## Description:

To enable debug_mode for specific Controller and then to view additional info in API Log
we refactored API LOG Config ActiveAdmin page (add update, create, delete actions).

So to view additional info in API Log first of all you should:
1. create new API log config (System->Api Log Config) for example: `Api::Rest::Admin::AuthController`
2. make request to `Api::Rest::Admin::AuthController` and then in Api log you will see additional data as (RESPONSE BODY, RESPONSE HEADERS, REQUEST BODY, REQUEST HEADERS)

## Changes

- Removed `bebug` column from `sys.api_log_config` table. Because in the case when the record exists in table `sys.api_log_config` it is already means that debug mode is enabled for specific controller.
- Implemented a separate service that returns all controller names within "Api" namespace.

## Screenshots

Screenshot 2023-10-05 at 16 09 28

### Related links
closes #1349 
documentation https://github.com/yeti-switch/doc/pull/133

### CODE REVIEW

**Comment by dmitry-sinina** [comment]
why we need "edit" there?

**Comment by Ivanov-Anton** [comment]
> why we need "edit" there?

@dmitry-sinina Ok I will remove the edit action, it will be all suggestions from you or something else?

File: app/services/return_api_controllers.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class ReturnApiControllers < ApplicationService
+  def call
+    controller_lists = []
+
+    Rails.application.routes.routes.each do |route|
```
[the remark this answers is not in the document]
Comment (Ivanov-Anton) [reply app/services/return_api_controllers.rb]: I think with local variable will be the same performance problem, each time instance of the class will be created and do not use previous results of found controller names. We can try to cache results with class variable.

[the remark this answers is not in the document]
Comment (Ivanov-Anton) [reply app/services/return_api_controllers.rb]: @dmitry-sinina what do you think about this idea.

File: app/controllers/api_controller.rb [hunk 2]
```diff
@@ -24,10 +24,7 @@ def db_logging
   end
 
   def debug_mode
-    unless instance_variable_defined?(:@debug_mode)
-      @debug_mode ||= System::ApiLogConfig.where(controller: self.class.name).pluck(:debug).first
-    end
-    @debug_mode
+    @debug_mode ||= System::ApiLogConfig.find_by(controller: self.class.name).present?
```
Comment (gigorok) [inline app/controllers/api_controller.rb:27]:
```
define_memoizable :debug_mode, apply: -> do
    System::ApiLogConfig.exist?(controller: self.class.name)
end
```

File: app/services/return_api_controllers.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class ReturnApiControllers < ApplicationService
```
Comment (gigorok) [inline app/services/return_api_controllers.rb:3]: use next module instead

```
module ApiControllers
  mattr_accessor :list

  def list
    return @list if @list

    @list = Rails.application.routes.set.anchored_routes.map(&:defaults).filter_map do |route|
      if route[:controller].start_with?('api/')
        "#{route[:controller].camelize}Controller}"
      end
    end
  end

  module_function :list
end
```

**Pull request merged**